dali/test/python/auto_aug/test_augmentation_decorator.py

```diff
@@ -0,0 +1,392 @@
+                                   num_magnitude_bins=11)
+        return const_mag, dyn_mag
+
+    p = pipeline(num_threads=4, device_id=0, batch_size=batch_size)
+    p.build()
+    const_mag, dyn_mag = p.run()
+    const_mag_ref = ref_param(mag_range, 5, [const_bin] * batch_size)
+    dyn_mag_ref = ref_param(mag_range, 11, list(range(batch_size)))
+    check_batch(const_mag, const_mag_ref, max_allowed_error=0)
+    check_batch(dyn_mag, dyn_mag_ref, max_allowed_error=0)
+
+
+def test_explicit_mag_range():
+    mag_range = np.array([1, 1, 2, 3, 5, 8, 13, 21])
+    batch_size = 8
+    const_bin = 7
+
+    @augmentation(mag_range=mag_range)
+    def pass_through_mag(sample, param):
+        return param
+
+    @pipeline_def
+    def pipeline():
+        idx_in_batch = sample_info(lambda info: info.idx_in_batch)
+        const_mag = pass_through_mag(types.Constant(42), magnitude_bin=const_bin)
+        dyn_mag = pass_through_mag(types.Constant(42), magnitude_bin=idx_in_batch)
+        return const_mag, dyn_mag
+
+    p = pipeline(num_threads=4, device_id=0, batch_size=batch_size)
+    p.build()
+    const_mag, dyn_mag = p.run()
+    const_mag_ref = ref_param(mag_range, None, [const_bin] * batch_size)
+    dyn_mag_ref = ref_param(mag_range, None, list(range(batch_size)))
+    check_batch(const_mag, const_mag_ref, max_allowed_error=0)
+    check_batch(dyn_mag, dyn_mag_ref, max_allowed_error=0)
+
+
+@params((((201, 260), 60, False, 0)), (((301, 330), 30, True, 29)), (((101, 150), 50, False, None)),
+        (((701, 710), 10, True, None)))
+def test_randomly_negate(mag_range, num_magnitude_bins, use_implicit_sign, const_mag):
+    batch_size = 64
+
+    @augmentation(mag_range=mag_range, randomly_negate=True)
+    def pass_through_mag(sample, param):
+        return param
+
+    @pipeline_def
+    def pipeline():
+        mag_sign = None if use_implicit_sign else sample_info(lambda info: info.idx_in_batch % 2)
+        magnitude_bin = const_mag if const_mag is not None else sample_info(
+            lambda info: info.idx_in_batch % num_magnitude_bins)
+
+        return pass_through_mag(types.Constant(42), magnitude_bin=magnitude_bin,
+                                num_magnitude_bins=num_magnitude_bins, random_sign=mag_sign)
+
+    if not use_implicit_sign:
+        p = pipeline(num_threads=4, device_id=0, batch_size=batch_size)
+    else:
+        warn_glob = "The augmentation `pass_through_mag` was * called without `random_sign`"
+        with assert_warns(Warning, glob=warn_glob):
+            p = pipeline(num_threads=4, device_id=0, batch_size=batch_size)
+    p.build()
+    magnitudes, = p.run()
+    magnitudes = [np.array(el) for el in magnitudes]
+    if use_implicit_sign:
+        # the implicit sign is random, just sanity check if
+        # there are some positive and negative magnitudes
+        assert any(el < 0 for el in magnitudes)
+        assert any(el > 0 for el in magnitudes)
```

Comment:
I guess you will get a fully positive or negative batch at some point, unless the seed will be fixed.

Reply:
Spot-on, I've added the seeds